x-pack/elastic-agent/pkg/agent/application/upgrade/service_darwin.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build darwin
+
+package upgrade
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/application/paths"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/errors"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/install"
+)
+
+const (
+	// delay after agent restart is performed to allow agent to tear down all the processes
+	// important mainly for windows, as it prevents removing files which are in use
+	afterRestartDelay = 2 * time.Second
+)
+
+// Init initializes os dependent properties.
+func (ch *CrashChecker) Init(ctx context.Context) error {
+	ch.sc = &pidProvider{}
+
+	return nil
+}
+
+type pidProvider struct{}
+
+func (p *pidProvider) Close() {}
+
+func (p *pidProvider) PID(ctx context.Context) (int, error) {
+	piders := []func(context.Context) (int, error){
+		// list of services differs when using sudo and not
+		// agent should be included in sudo one but in case it's not
+		// we're falling back to regular
+		p.piderFromCmd(ctx, "sudo", "launchctl", "list", install.ServiceName),
+		p.piderFromCmd(ctx, "launchctl", "list", install.ServiceName),
```

Comment:
@blakerouse @michalpristas I've seen the `sudo` reference is this really the way to go, is it possible that it fails?

Reply:
@ph I didn't notice the `sudo`. `sudo` should be removed, the Agent is already running at high-level permissions if its running under the service manager, which it must for self-upgrading.

Reply:
i dont know why but when i run it without sudo i dont see elastic-agent service listed

Reply:
edit; workin when running as a service, without sudo it wont work  when using DEV build and trying upgrades, i will add DEV check and perform sudo only in this case. otherwise sudoless command